src/message/__tests__/fetchActions-test.js

```diff
@@ -342,46 +342,33 @@ describe('fetchActions', () => {
       test("rejects when validation-at-the-edge can't handle data, dispatches MESSAGE_FETCH_ERROR", async () => {
         // This validation is done in transformFetchedMessages in
         // rawModelTypes.
-        //
-        // Simulate #4156, a real-life problem that a user at server commit
-        // 0af2f9d838 ran into [1], by having `user` be missing on reactions
-        // on a message:
-        //   https://github.com/zulip/zulip-mobile/issues/4156#issuecomment-655905093
         const store = mockStore<GlobalState, Action>(baseState);
 
-        // Missing `user` (and `user_id`, for good measure), to
-        // simulate #4156.
-        const faultyReaction = {
```

Comment:
I think the comment "Simulate #…, a real-life problem that a user ran into [link]" is helpful — this is otherwise a pretty weird-looking test.

Given that context, and given this latter comment too ­— which says `user_id` is missing for good measure — what happens if we leave the test unchanged, with `user_id` still missing? … Ah, I guess the point is that this is testing what happens if `transformFetchedMessage` (or anything else within the `migrateResponse` helper of `api.getMessages`) throws an exception. Which doesn't happen for just any malformed data, only for malformations in the data it actually has any transformations to perform on.